python_modules/dagit/bin/dagit

```diff
@@ -1,19 +1,41 @@
-#!/bin/bash
-ARGS=()
-WATCH="0"
-for var in "$@"; do
-    if [ "$var" == '--no-watch' ]; then
-      WATCH="0"
-    if [ "$var" == '--watch' ]; then
-      WATCH="1"
-    else
-      ARGS+=("$var")
-    fi
-done
+#!/usr/bin/env python
+import sys
+import time
+import signal
+from watchdog.observers import Observer
+from watchdog.tricks import AutoRestartTrick
 
-if [ "$WATCH" = "1" ]; then
-  echo "Watching for file changes..."
-  watchmedo auto-restart -p="*.py" -R dagit-cli -- "$@"
-else
-  dagit-cli "${ARGS[@]}"
-fi
+# Build the dagit-cli command, omitting the --no-watch arg if present
+watch = True
+command = ["dagit-cli"]
+for arg in sys.argv[1:]:
+    if arg == "--no-watch":
+        watch = False
+    else:
+        command.append(arg)
+
+handler = AutoRestartTrick(command=command,
```

Comment:
I think you should leave a comment here that is essentially the description in your PR summary. With a pointer to the original code and why you copy/pasted it here with the modes you made. Also maybe call out the changed lines.

Reply:
👍